### PR TITLE
Unlimited number of poll options

### LINKED ISSUE #1272: Support displaying polls with more than 4 options
Some federated software like Misskey (who originally implemented polls) and soon Pleroma allow for more than 4 options in polls, however when trying to preview a post with more than 4 options in Tusky it just shows the first 4 of them. 

Mastodon does not reject/trim options on the backend and properly displays more than 4 of them in the frontend so it would be nice if Tusky displayed them all too.

* * * *
- Tusky Version: 7.0
- Android Version: N/A
- Android Device: N/A
- Mastodon instance (if applicable): N/A

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

### PR DESCRIPTION
**Pull request by connyduck**
- support unlimited number of poll options 
- voting selection no longer gets lost when scrolling
- fix crash when poll has no end date

This makes the code much shorter, I should have done it like this from the beginning 🙄 

closes #1272

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/adapter/PollAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,108 @@
+package com.keylesspalace.tusky.adapter
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/adapter/PollAdapter.kt]: Missing license/copyright (and PollViewData too)

File: app/src/main/java/com/keylesspalace/tusky/adapter/PollAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        this.pollOptions = options
+        this.voteCount = voteCount
+        this.emojis = emojis
+        this.mode = mode
+        notifyDataSetChanged()
+    }
+
+    fun getSelected() : List<Int> {
+        return pollOptions.filter { it.selected }
+                .map { pollOptions.indexOf(it) }
```
[the remark this answers is not in the document]
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/adapter/PollAdapter.kt]: yes, I thought about this too, but I need the index before the filtering so either this or mapping to `Pair` first

File: app/src/main/java/com/keylesspalace/tusky/adapter/PollAdapter.kt (hunk 1 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/adapter/PollAdapter.kt]: thx!

File: app/src/main/java/com/keylesspalace/tusky/adapter/PollAdapter.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/adapter/PollAdapter.kt]: ah, ofc, sorry!

**Pull request merged**